core/src/main/java/org/fenixedu/messaging/domain/MessageTask.java

```diff
@@ -17,5 +16,6 @@ public void runTask() {
             dispatched++;
         }
         taskLog("Dispatched %d messages\n", dispatched);
+        MessagingSystem.deleteOldMessages();
```

Comment:
To make the quantity message deletion policy more consistent. Well, assuming the dispatching is synchronous, this makes it consistent. Otherwise it's about the same.
Scenario: person has 5 messages sent, queues 7 more, while the policy is to keep 10 sent. By deleting first people are more likely to see 12 sent messages in their sender when the limit says 10 on the same page. "Oh how relieved I am this message is still available!" / 'goes away for 5 minutes / 'task runs / 'comes back / "My precious message is gone!!" / Ticketmageddon ensues.